Add docker image for secured version and README with the details

dlumbrer opened the pull request:
This commit adds the needed files for the image creation process of the 6.8.6 Kibiter secured version. For Bitergia purposes.



<!--
Thank you for your interest in and contributing to Kibana! There
are a few simple things to check before submitting your pull request
that can help with the review process. You should delete these items
from your submission, but they are here to help bring them to your
attention.

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md)?
- If submitting code, have you included unit tests that cover the changes?
- If submitting code, have you tested and built your code locally prior to submission with `npm test && npm run build`?
- If submitting code, is your pull request against master? Unless there is a good reason otherwise, we prefer pull requests against master and will backport as needed.
-->

dlumbrer committed:
[docker] Add Dockerfile and docker entrypoint files for 6.8.6

This commit adds the needed files for the image creation process
of the 6.8.6 Kibiter secured version. For Bitergia purposes.

Signed-off-by: David Moreno <dmoreno@bitergia.com>

sduenas commented on docker/Dockerfile:
```diff
@@ -0,0 +1,120 @@
+
+# backports used to install java 8 needed by elasticsearch 5
+# ruby gems are used for building binary packages
+RUN apt-get update && \
```
Do you think it makes sense to have all the installation in the same step? I think it will be better to split it in several parts to improve the speed in case the image fails in this step.

dlumbrer replied:
I agree, thanks!

sduenas reviewed (changes requested):
Would you mind to add a README file with the env variables you defined and how to use them? Please also check my other comment about the dockerfile.

dlumbrer committed:
Add README with the information about Kibiter 6.8.6

This commit adds the README with all the information about the
6.8.6 version of Kibiter, including the docker installation,
the source code installation, the docker env variables
description, the features and the community guidelines to open
issues, contact, etc.

Signed-off-by: David Moreno <dmoreno@bitergia.com>

dlumbrer commented:
Updates @sduenas, thanks

sduenas reviewed (changes requested):
LGTM but I found the next error building the image:

```
Step 25/30 : RUN ./opt/kibana/bin/kibana | sleep 200 > /tmp/run_kibana_log.txt
 ---> Running in 4af05fbcda26
Browserslist: caniuse-lite is outdated. Please run next command `npm update caniuse-lite browserslist`


events.js:174
      throw er; // Unhandled 'error' event
      ^

Error: write EPIPE
    at WriteWrap.afterWrite (net.js:782:14)
Emitted 'error' event at:
    at Socket.onerror (_stream_readable.js:691:12)
    at Socket.emit (events.js:194:15)
    at onwriteError (_stream_writable.js:431:12)
    at onwrite (_stream_writable.js:456:5)
    at _destroy (internal/streams/destroy.js:40:7)
    at Socket.dummyDestroy [as _destroy] (internal/process/stdio.js:11:34)
    at Socket.destroy (internal/streams/destroy.js:32:8)
    at WriteWrap.afterWrite (net.js:784:10)
```

sduenas reviewed (commented):
While running the container I also found the next error, so the container fails before it starts:

```
kibiter_1        | /docker_entrypoint.sh: line 67: syntax error: unexpected end of file
```
This is part of the docker-compose I use to run the container:
```
kibiter:
  restart: on-failure:5
  image: integration-6.8.6
  environment:
    - PROJECT_NAME=Development
    - NODE_OPTIONS=--max-old-space-size=1000
    - ELASTICSEARCH_URL=https://admin:admin@elasticsearch:9200
    - ELASTICSEARCH_USER=admin
    - ELASTICSEARCH_PASSWORD=admin
  links:
    - elasticsearch
  ports:
    - 5601:5601

```

dlumbrer commented:
> LGTM but I found the next error building the image:
> 
> ```
> Step 25/30 : RUN ./opt/kibana/bin/kibana | sleep 200 > /tmp/run_kibana_log.txt
>  ---> Running in 4af05fbcda26
> Browserslist: caniuse-lite is outdated. Please run next command `npm update caniuse-lite browserslist`
> 
> 
> events.js:174
>       throw er; // Unhandled 'error' event
>       ^
> 
> Error: write EPIPE
>     at WriteWrap.afterWrite (net.js:782:14)
> Emitted 'error' event at:
>     at Socket.onerror (_stream_readable.js:691:12)
>     at Socket.emit (events.js:194:15)
>     at onwriteError (_stream_writable.js:431:12)
>     at onwrite (_stream_writable.js:456:5)
>     at _destroy (internal/streams/destroy.js:40:7)
>     at Socket.dummyDestroy [as _destroy] (internal/process/stdio.js:11:34)
>     at Socket.destroy (internal/streams/destroy.js:32:8)
>     at WriteWrap.afterWrite (net.js:784:10)
> ```

This is a "controlled" error because we force to init Kibiter in the image creation process, forcing Kibiter to bundle its scripts (optimize process) and then launching the next time instantly. The error appears because we "kill" Kibiter.


 

> While running the container I also found the next error, so the container fails before it starts:
> 
> ```
> kibiter_1        | /docker_entrypoint.sh: line 67: syntax error: unexpected end of file
> ```
> 
> This is part of the docker-compose I use to run the container:
> 
> ```
> kibiter:
>   restart: on-failure:5
>   image: integration-6.8.6
>   environment:
>     - PROJECT_NAME=Development
>     - NODE_OPTIONS=--max-old-space-size=1000
>     - ELASTICSEARCH_URL=https://admin:admin@elasticsearch:9200
>     - ELASTICSEARCH_USER=admin
>     - ELASTICSEARCH_PASSWORD=admin
>   links:
>     - elasticsearch
>   ports:
>     - 5601:5601
> ```

I solve it, I added by mistake an "else" instead of a "fi", sorry about that

sduenas reviewed (commented):
Tested. It works perfectly. You can merge it.
